stl/inc/xutility

```diff
@@ -4729,8 +4725,8 @@ _NODISCARD _CONSTEXPR20 bool equal(const _InIt1 _First1, const _InIt1 _Last1, co
     return true;
 }
 #else // ^^^ _HAS_IF_CONSTEXPR / !_HAS_IF_CONSTEXPR vvv
-template <class _InIt1, class _InIt2, class _Pr>
-bool _Equal_unchecked1(_InIt1 _First1, const _InIt1 _Last1, _InIt2 _First2, _Pr _Pred, false_type) {
+template <class _InIt1, class _InIt2, class _Pr, enable_if_t<!_Equal_memcmp_is_safe<_InIt1, _InIt2, _Pr>, int> = 0>
+bool _Equal_unchecked(_InIt1 _First1, const _InIt1 _Last1, _InIt2 _First2, _Pr _Pred) {

```

Comment:
Note: the `!_HAS_IF_CONSTEXPR` region is correctly transformed, because these are the `_Equal_unchecked` helper functions. We've already performed unwrapping in `equal()`. (This structure was necessary in the era before `decltype` when we weren't able to ask "what is the type of this iterator after unwrapping" except by introducing a helper function to perform template argument deduction, which then needed another layer of helper functions to perform tag dispatch.)

No change requested here.